File: c/easy/minimum_changes_to_make_alternating_binary_string.c
#include <string.h>

int minOperations(const char *s) {
    int mismatches_starting_with_zero = 0;
    int length = strlen(s);

    for (int index = 0; index < length; index++) {
        mismatches_starting_with_zero += ((s[index] & 1) ^ index) & 1;
    }

    int mismatches_starting_with_one = length - mismatches_starting_with_zero;

    return mismatches_starting_with_zero < mismatches_starting_with_one
        ? mismatches_starting_with_zero
        : mismatches_starting_with_one;
}

File: rust/easy/minimum_changes_to_make_alternating_binary_string.rs
pub fn min_operations(s: &str) -> i32 {
    let mut mismatches_starting_with_zero = 0;

    for (index, byte) in s.bytes().enumerate() {
        mismatches_starting_with_zero += (((byte & 1) ^ (index as u8)) & 1) as i32;
    }

    let length = s.len() as i32;
    let mismatches_starting_with_one = length - mismatches_starting_with_zero;

    mismatches_starting_with_zero.min(mismatches_starting_with_one)
}